### PR TITLE
Watchdog timer

### PR DESCRIPTION
**Pull request by PixelToast**

### CODE REVIEW

File: src/watchdog.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+// iwdgStart - Initializes the watchdog with a timeout
+void iwdgStart() {
+	if (!iwdgEnabled) return;
+
+	uint16_t prescalerCode = 0;
+	uint16_t prescaler = 4;
+
+	while (IWDG_TIMEOUT * (40 / prescaler) >= 0x7FF && prescalerCode < 7) {
```
Comment (edjubuh) [inline src/watchdog.c]: This prescalerCode and prescaler are always the same for end users. This computation shouldn't be performed every time.

Comment (PixelToast) [inline src/watchdog.c]: My bad i thought gcc could optimize that out, i checked the disassembly and it looks like its still there:
`00000000 <iwdgStart>:
   0:	4b16      	ldr	r3, [pc, #88]	; (5c <iwdgStart+0x5c>)
   2:	b470      	push	{r4, r5, r6}
   4:	781b      	ldrb	r3, [r3, #0]
   6:	b33b      	cbz	r3, 58 <iwdgStart+0x58>
   8:	f240 76fe 	movw	r6, #2046	; 0x7fe
   c:	2100      	movs	r1, #0
   e:	2304      	movs	r3, #4
  10:	2428      	movs	r4, #40	; 0x28
  12:	f44f 757a 	mov.w	r5, #1000	; 0x3e8
  16:	fb94 f2f3 	sdiv	r2, r4, r3
  1a:	436a      	muls	r2, r5
  1c:	42b2      	cmp	r2, r6
  1e:	b288      	uxth	r0, r1
  20:	dd06      	ble.n	30 <iwdgStart+0x30>
  22:	3101      	adds	r1, #1
  24:	2908      	cmp	r1, #8
  26:	d002      	beq.n	2e <iwdgStart+0x2e>
  28:	005b      	lsls	r3, r3, #1
  2a:	b29b      	uxth	r3, r3
  2c:	e7f3      	b.n	16 <iwdgStart+0x16>
  2e:	2007      	movs	r0, #7
  30:	f245 5155 	movw	r1, #21845	; 0x5555
  34:	bc70      	pop	{r4, r5, r6}
  36:	4b0a      	ldr	r3, [pc, #40]	; (60 <iwdgStart+0x60>)
  38:	6019      	str	r1, [r3, #0]
  3a:	6058      	str	r0, [r3, #4]
  3c:	f64c 41cc 	movw	r1, #52428	; 0xcccc
  40:	609a      	str	r2, [r3, #8]
  42:	f64a 22aa 	movw	r2, #43690	; 0xaaaa
  46:	4807      	ldr	r0, [pc, #28]	; (64 <iwdgStart+0x64>)
  48:	601a      	str	r2, [r3, #0]
  4a:	6019      	str	r1, [r3, #0]
  4c:	601a      	str	r2, [r3, #0]
  4e:	2140      	movs	r1, #64	; 0x40
  50:	2302      	movs	r3, #2
  52:	2200      	movs	r2, #0
  54:	f7ff bffe 	b.w	0 <taskCreate>
  58:	bc70      	pop	{r4, r5, r6}
  5a:	4770      	bx	lr
  5c:	00000000 	andeq	r0, r0, r0
  60:	40003000 	andmi	r3, r0, r0
  64:	00000000 	andeq	r0, r0, r0
`

**Comment by PixelToast** [comment]
Referencing #6

**Comment by edjubuh** [comment]
Works with IME static discharge

**Pull request merged**